### PR TITLE
chore: push eol notification for macOS 10.11 and 10.12

### PR DESCRIPTION
**Pull request by deepak1556**
Refs https://github.com/microsoft/vscode/issues/123374#issuecomment-1385277353

Screenshot 2023-01-18 at 19 06 50

### CODE REVIEW

**Comment by deepak1556** [comment]
@isidorn need your input on the message and also learn more links are pending the website changes.

**Comment by isidorn** [comment]
@deepak1556 thanks for this PR.
The message looks good. Here's some minor polish:
OLD: "{0} on {1} will stop receiving updates soon"
NEW: "{0} on {1} will soon stop receiving updates. Consider upgrading your macOS version."

I have just created a docs PR for the website changes https://github.com/microsoft/vscode-docs/pull/5943/files

**Comment by deepak1556** [comment]
Updated, pending new faq links.

**Comment by isidorn** [comment]
@deepak1556 do you want me to request new aka.ms links (I forgot how to do this but can find out)? We need something like:
https://aka.ms/vscode-faq-old-macOS
That points to 
https://code.visualstudio.com/docs/supporting/faq#_can-i-run-vs-code-on-old-macos-versions

File: src/vs/workbench/electron-sandbox/window.ts [hunk 1]
```diff
@@ -719,6 +719,43 @@ export class NativeWindow extends Disposable {
...
+					actions,
+					icon: Codicon.warning
+				});
+
+				this.notificationService.prompt(
+					Severity.Warning,
+					message,
+					[{
+						label: localize('learnMore', "Learn More"),
+						run: () => this.openerService.open(URI.parse('https://aka.ms/vscode-faq-win7'))
```
Comment (bpasero) [inline src/vs/workbench/electron-sandbox/window.ts:750]: Do we need a new link?

File: src/vs/workbench/electron-sandbox/window.ts [hunk 2]
```diff
@@ -719,6 +719,43 @@ export class NativeWindow extends Disposable {
...
+			const eolReleases = new Map<string, string>([
+				['15', 'OS X El Capitan'],
+				['16', 'macOS Sierra'],
+			]);
+			// Refs https://en.wikipedia.org/wiki/Darwin_%28operating_system%29#Release_history
+			if (eolReleases.has(majorVersion)) {
+				const message = localize('macoseolmessage', "{0} on {1} will soon stop receiving updates. Consider upgrading your macOS version.", this.productService.nameLong, eolReleases.get(majorVersion));
+				const actions = [{
+					label: localize('macoseolBannerLearnMore', "Learn More"),
+					href: 'https://aka.ms/vscode-faq-win7'
```
Comment (bpasero) [inline src/vs/workbench/electron-sandbox/window.ts:734]: Also here

**Comment by deepak1556** [comment]
@isidorn I have created the link but have not made it active. Let me know once the page is live and I can activate the link.

File: src/vs/workbench/electron-sandbox/window.ts (hunk 1 above)
Comment (deepak1556) [reply src/vs/workbench/electron-sandbox/window.ts:750]: Yup it was pending faq updates in the website.

**Review by bpasero — APPROVED** [review]
lgtm, if we want to merge it now, then we need a version check of electron I suppose

**Comment by deepak1556** [comment]
@bpasero can you clarify, why do we need an electron version check ?

**Comment by bpasero** [comment]
@deepak1556 oh sorry, nevermind. Yes, we want this to ship now even before updating.

**Comment by isidorn** [comment]
@deepak1556 ok, will ping you once the change goes live on our website. Thank you

**Pull request merged**